src/Collections-Unordered/Bag.class.st

```diff
@@ -2,6 +2,39 @@
 I represent an unordered collection of possibly duplicate elements.
 	
 I store these elements in a dictionary, tallying up occurrences of equal objects. Because I store an occurrence only once, my clients should beware that objects they store will not necessarily be retrieved such that == is true. If the client cares, a subclass of me should be created.
+
+## Creating 
+Bags can be created from other collections or built up progressively using the `Bag>>#add:` message. The example below shows four different ways to create a `Bag` with 3 occurrences of 1.
+```
+b := #(1 1 1) asBag.
+""or""
+b:= Bag newFrom: #(1 1 1).
+""or""
+b := Bag new add: 1; add: 1; add: 1.
+""or"" 
+b add: 1 withOccurrences: 3.
+
+b valuesAndCounts.  ""Dictionary(1->3)""
+```
+
+## Enumerating
+
+It is possible to iterate over each individual occurrance of an object using the `Bag>>#do:` message, or pairs of objects and their occurances using `Bag>>#doWithOccurrences:` or the synonymous messages `Bag>>#associationsDo:` and `Bag>>#keysAndValuesDo:`.
+
+```
+b := #(1 1 1 1) asBag.
+b do: [:obj | obj].  ""will be 1. 1. 1. 1."" 
```

Comment:
I would use do: with a side effect

| nbElts |
nbElts := 0.
b := #(1 1 1) asBag.
b do: [:each | nbElts := nbElts + 1].

Reply:
Yes, this wasn't clear. I've updated the example to use a counter as you suggest.